### DISCUSSION TITLE
Question about DNS forwarding

### DISCUSSION THREAD

**Discussion by Horizonbli** [celzero/rethink-app#537] [Q&A]
I have RethinkDNS set to forward dns queries to a local proxy, which on its turn has a defined set of block rules, which RethinkDNS on its turn shows in its own UI that DNS queries are either allowed or blicked (by the local proxy).

My question is: Does RethinkDNS own blocking list have any effect if RDNS is set to forward DNS queries to a local proxy or is it useless to enable it?

By the way, a "deny list" for testing purposes would be great. I don't think there is one?

Thanks

**Comment by ignoramous** [comment]
> By the way, a "deny list" for testing purposes would be great. I don't think there is one?

Not just for testing, but a custom deny list would be great to have! We want to implement this on priority but only after we are done with `v054` (unblock individual domains) and `v055` (per-app WireGuard, per-app Proxies, per-app DNS).

**Reply by ignoramous** [reply]
> My question is: Does RethinkDNS own blocking list have any effect if RDNS is set to forward DNS queries to a local proxy or is it useless to enable it?

Yes, starting `v053i` Rethink honors local blocklists (if set) before forwarding the DNS query/answer.

(Since this is a question, I have moved it to _discussions_).
